internal/services/redis/redis_cache_data_source.go

```diff
@@ -344,6 +349,8 @@ func dataSourceRedisCacheRead(d *pluginsdk.ResourceData, meta interface{}) error
 		d.Set("primary_connection_string", getRedisConnectionString(*props.HostName, *props.SslPort, *keys.Model.PrimaryKey, enableSslPort))
 		d.Set("secondary_connection_string", getRedisConnectionString(*props.HostName, *props.SslPort, *keys.Model.SecondaryKey, enableSslPort))
 
+		d.Set("access_keys_authentication_enabled", !(*props.DisableAccessKeyAuthentication))
```

Comment:
Shall we nil check and provide a default value if nil:

```suggestion
                accessKeyAuthEnabled := true
                if props.DisableAccessKeyAuthentication != nil {
                    accessKeyAuthEnabled = !(*props.DisableAccessKeyAuthentication)
                }
		d.Set("access_keys_authentication_enabled", accessKeyAuthEnabled)
```